spec.bs

```diff
@@ -1667,12 +1733,22 @@ successful [=navigate|navigation=] to a [=top-level traversable=].
 </div>
 
 <div algorithm="automatic beacon patch">
-  Modify [[HTML]]'s [=attempt to populate the history entry's document=] algorithm. In step 6,
-  substep 11, add a new step after step 5 that reads:
+  Modify [[HTML]]'s [=attempt to populate the history entry's document=] algorithm in the following ways.
+
+  First, add steps after step 5 that reads:
+
+  6. Let |attributionReportingEnabled| be the result of determining whether |navigable|'s
```

Comment:
This is still running in parallel, and before the new document has been installed in the target navigable/browsing context. So you're pulling the "old" origin and also checking `[=allowed to use=]` in the old document. Is that intended?

Reply:
Yeah, that's intended to check the initiator origin, therefore the old origin.

Reply:
Ah, well the initiator origin may not be the "old document" origin, right? Any frame might have initiated this document. So for any navigation there are three possible origins:
 - The initiator origin
 - The target frame's old document origin
 - The target document's _new_ origin

Which do you want here?

Reply:
We want the initiator origin, and check [=allowed to use=] on the initiator frame. What's the proper way to get those?

Reply:
You'll want to grab source snapshot params's fetch client's [origin](https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-origin) for this. Although, I'm curious: what does the implementation do if that initiator document is already gone by the time we make it here?

Reply:
Or, I guess you'll want the "associated Document", not the "origin"